OSDOCS-18308: Created Autonode documentation for ROSA (#110935)
openshift/openshift-docs: modules/rosa-nodes-autonode-managing-enable-ui.adoc

.Procedure
. Log in to the AWS console. 
. In the AWS console, navigate to **IAM > Roles**.
. On your {autonode} ARN, update the trust policy to the following:
+
[NOTE]
====
Do not include the `https://` prefix from the OIDC Endpoint URL. For example, use  `d3gt1gce2zmg3d.cloudfront.net/22qa79chsq8mand8hvmnr33upj48lmas` instead of `https://d3gt1gce2zmg3d.cloudfront.net/22qa79chsq8mand8hvmnr33upj48lmas`.
====
+
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Principal": {
                "Federated": "arn:aws:iam::720424066366:oidc-provider/<oidc-endpoint-url>"
            },
            "Action": "sts:AssumeRoleWithWebIdentity",
            "Condition": {
                "StringEquals": {
                    "<oidc-endpoint-url>:sub": "system:serviceaccount:kube-system:karpenter"
                }
            }
        }
    ]
}
```
. In {cluster-manager-url}, select your cluster from the cluster list.
. On the cluster details screen, select the Pencil icon next to the status for {autonode}. 
. On the *Edit autonode settings* dialog box, toggle "Enable Autonode".
+
[IMPORTANT]
====

> Once you enable {autonode} on your cluster, you cannot disable this feature. You must create a new cluster.

Reviewer: You can leave out 'You must create a new cluster.' 

1. It's implied that when something is not supported, or immutable, a new resource must be created.
2. It's technically possible to remove the resources so 'you must create a new cluster' is not techically accurate 100% if customer only wants to not have karpenter manage any node.

.Procedure

. Export your AWS ID:
+
[source,terminal]
----
$ export AWS_ACCOUNT_ID=$(aws sts get-caller-identity --query Account --output text)
----

. Log in to the AWS console. 
. In the link:https://us-east-2.console.aws.amazon.com/iam/home?region=us-east-2#/roles[AWS console], navigate to **IAM > Roles**.
. On your {autonode} ARN, update the trust policy to include the following policy specifications:
+
[NOTE]
====
To access this ARN, run:

[source,terminal]
----
$ echo $ROLE_ARN
----
====
+
[source,terminal]
----
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Principal": {
                "Federated": "arn:aws:iam::${AWS_ACCOUNT_ID}:oidc-provider/<oidc-endpoint-url>"
            },
            "Action": "sts:AssumeRoleWithWebIdentity",
            "Condition": {
                "StringEquals": {
                    "<oidc-endpoint-url>:sub": "system:serviceaccount:kube-system:karpenter"
                }
            }
        }
    ]
}
----

. In {cluster-manager-url}, select your cluster from the cluster list.
. On the cluster details screen, select the Pencil icon next to the status for {autonode}. 

> . On the *Edit autonode settings* dialog box, toggle "Enable Autonode".

Reviewer 2: Maybe add something after line 70 to include a step of entering the arn into the `Autonode IAM role ARN` input

// Module included in the following assemblies:
//
// * rosa_cluster_admin/rosa-nodes-managing-karpenter.adoc

:_mod-docs-content-type: PROCEDURE
[id="rosa-nodes-autonode-managing-enable-ui_{context}"]

> = Enabling {autonode} using {cluster-manager}

Reviewer 3: ```suggestion
= Enable {autonode} using {cluster-manager}
```

[role="_abstract"]
Enable {autonode} on your cluster by using {cluster-manager} after it finishes installing.

.Prerequisites
* You have created a {product-title} cluster, version 4.22.0 or later.
* You have created an IAM role to be configured for {autonode}.

> * You have your cluster's OIDC Endpoint URL.

Reviewer 3: In the ideal world, IAM and OIDC should be expanded.
Probabl expanding URL here will make the text heavy, up to you.

+ 
[NOTE]
====
Run `rosa describe cluster -c $CLUSTER_NAME | grep "OIDC Endpoint URL"` to see this URL. Do not include the `https://` prefix from the OIDC Endpoint URL. For example, use  `example-oidc-endpoint.cloudfront.net/abcd1234examplehash5678` instead of `https://example-oidc-endpoint.cloudfront.net/abcd1234examplehash5678`.
====
* You have the proper credentials to access the AWS console.

.Procedure

. Export your AWS ID:
+
[source,terminal]
----
$ export AWS_ACCOUNT_ID=$(aws sts get-caller-identity --query Account --output text)
----

. Log in to the link:https://console.aws.amazon.com/[AWS console]. 
. In the AWS console, navigate to **IAM > Roles**.

> . On your {autonode} ARN, update the trust policy to include the following policy specifications:

Reviewer 3: ```suggestion
. On your {autonode} Amazon Resource Name (ARN), update the trust policy to include the following policy specifications:
```

+
[NOTE]
====
To access this ARN, run:

[source,terminal]
----
$ echo $ROLE_ARN
----
====
+
[source,json]
----
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Principal": {
                "Federated": "arn:aws:iam::<aws_account_id>:oidc-provider/<oidc-endpoint-url>"
            },
            "Action": "sts:AssumeRoleWithWebIdentity",
            "Condition": {
                "StringEquals": {
                    "<oidc-endpoint-url>:sub": "system:serviceaccount:kube-system:karpenter"
                }
            }
        }
    ]
}
----
where:
<aws_account_id>::
Use your AWS Account ID in this value. 
<oidc-endpoint-url>::

> Use the OIDC endpoint URL that you acquired.

Reviewer 3: ```suggestion
<aws_account_id>::
Specifies your AWS Account ID. 
<oidc-endpoint-url>::
Specifies the OIDC endpoint URL that you acquired.
```
Just a nitpick. This way we could be consistent with how the OCP docs work around callouts.
